src/main/java/org/candlepin/subscriptions/controller/TallyRetentionController.java

```diff
@@ -37,12 +41,26 @@
     private final TallySnapshotRepository repository;
     private final TallyRetentionPolicy policy;
 
-    public TallyRetentionController(TallySnapshotRepository repository, TallyRetentionPolicy policy) {
+    private final AccountListSource accountListSource;
+
+    public TallyRetentionController(TallySnapshotRepository repository, TallyRetentionPolicy policy,
+        AccountListSource accountListSource) {
         this.repository = repository;
         this.policy = policy;
+        this.accountListSource = accountListSource;
+    }
+
+    public void purgeSnapshots() {
+        try {
+            List<String> accountList = accountListSource.list();
+            accountList.forEach(this::cleanStaleSnapshotsForAccount);
+        }
+        catch (IOException e) {
+            throw new JobFailureException("Could not purge snapshots", e);
```

Comment:
I'm not sure that this should be throwing a JobFailureException. I'd like to see the job itself trap any exceptions from this method, then have it throw the JobFailureException.